blask-6/app/routes/blog.py

```diff
@@ -0,0 +1,33 @@
+from flask import Blueprint, render_template, redirect, url_for, request
+#last 3 imports are for the new form
+from app.models.blog import BlogPost, BlogPostForm
+
+# Create the blog Blueprint.  Note that all routes on the blog blueprint have
+# '/blog' prepended to them.
+blog = Blueprint('blog', __name__, url_prefix='/blog')
+
+@blog.route('/')  # Accessible at /blog/
+def blog_page():
+    """The blog page."""
+    #objects() method comes from the Flask mongoengine
+    posts = BlogPost.objects()
+    return render_template('blog.html', posts=posts)
+
+@blog.route('/new', methods=['GET', 'POST'])
+def new():
+    """Create a new post"""
+    form = BlogPostForm(request.form)
+
+    if request.method =='POST' and form.validate():
+        form.save()
```

Comment:
We should probably explain what `form.save()` does.  (explain that it creates an entry in Mongo)